shotover/src/hot_reload/protocol.rs

```diff
@@ -2,12 +2,17 @@
 
 use serde::{Deserialize, Serialize};
 use std::os::unix::io::OwnedFd;
+use std::time::Duration;
 
 /// Requests that can be sent from replacement shotover to original shotover
 #[derive(Debug, Serialize, Deserialize)]
 pub enum Request {
     SendListeningSockets,
-    GradualShutdown,
+    /// GradualShutdown request with the total duration for the shutdown process.
+    /// The duration specifies how long the original instance should spend draining connections.
+    GradualShutdown {
+        duration_secs: u64,
```

Comment:
does this work as a `Duration` instead of u64? It might not work because Serde might not support serializing/deserializing it, but worth checking.